Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/Armoury.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class Armoury : MonoBehaviour
+{
+    public Player player;
+    public PointsSystem currency;
+
+    IList<int> GetBuyableUnits; //Change to UnitType once Units are in
+    IList<int> GetBuyableCards;
+    bool b_IsCastlePieceBuyable;
+
+    IDictionary<int, int> UnitPrices { get; set; } //Change key to UnitType once Units are in
+    IDictionary<int, int> CardPrices { get; set; }
+    IDictionary<int, int> CastlePiecePrices { get; set; }
+
+    public delegate void PurchasedItemCallback(int id);
+    public event PurchasedItemCallback OnPurchasedItem = delegate { };
+
+    // Use this for initialization
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    public void PurchaseUnit(int unitId)
+    {
+        var unitPrice = UnitPrices.First(x => x.Key == unitId).Value;
+
+        if (GetBuyableUnits.Contains(unitId))
+        {
+            if (currency.getPoints() < unitPrice)
+            {
+                //Call UI, saying that player hasn't enough coins
+                //Return from function and don't call purchaseItem delegate
+                return;
+            }
+
+            //Give player the unit
+
+            //Remove currency
+            currency.removePoints(unitPrice, unitId.ToString());
+
+            //Call Delegate
+            var purchasedItem = new PurchasedItemCallback(PurchaseUnit);
```

Comment:
Here you can just call OnPurchasedItem(unitId); rather than creating a new object.